data/view.js

```diff
@@ -152,15 +166,19 @@ var WebSocketsView = createView(PanelView,
 
   onPrefChanged: function(event) {
     var prefName = event.prefName;
-    if (prefName != "tabularView") {
-      return;
-    }
 
-    // Update the way how frames are displayed.
-    if (event.newValue) {
-      store.dispatch(showTableView());
+    if (prefName === "tabularView") {
+
+      // Update the way how frames are displayed.
+      if (event.newValue) {
+        store.dispatch(showTableView());
+      } else {
+        store.dispatch(showListView());
+      }
     } else {
```

Comment:
Please add a condition checking the `prefName` and making sure that `updateConfig` is executed only for `enableSocketIo`, `enableSocketIo`, `enableJson` and `enhableMqtt`.

Honza